packages/vite/src/ember.ts

```diff
@@ -0,0 +1,52 @@
+import { templateTag } from './template-tag.js';
+import { resolver } from './resolver.js';
+import { mergeConfig, type UserConfig } from 'vite';
+import { esBuildResolver } from './esbuild-resolver.js';
+
+export let extensions = ['.mjs', '.gjs', '.js', '.mts', '.gts', '.ts', '.hbs', '.hbs.js', '.json'];
+
+export function ember() {
+  return [
+    templateTag(),
+    resolver(),
+    {
+      name: 'vite-plugin-ember-config',
+      async config(config: UserConfig, env: { mode: string; command: string }) {
+        if (config?.resolve?.extensions) {
+          if (extensions.some(ext => !config?.resolve?.extensions?.includes(ext))) {
+            throw new Error(
+              `You are customising your resolve.extensions but you are missign some that are required for Ember. Make sure you include at least all of the following ${extensions}`
+            );
+          }
+        }
+
+        return mergeConfig(config, {
```

Comment:
this should be the other way around, otherwise our config will override the user config.

Reply:
🤦 I read the docs like 20 times and convinced myself I did it the right way round... but you are of course correct, a simple test confirmed it 🫠 I've fixed it now